src/js/settings.js

```diff
@@ -591,33 +591,145 @@ function exportSubscriptions() {
   const dateYear = date.getFullYear();
   const dateString = 'freetube-subscriptions-' + dateYear + '-' + dateMonth + '-' + dateDay;
 
-  // Open user file browser. User gives location of file to be created.
+    switch(document.querySelector('#exportSelect').value){
+      
+      case "NewPipe":
+        exportNewpipeSubscriptions(dateYear, dateMonth, dateDay); 
+      break;
+      case "OPML":
+        exportOpmlSubscriptions(dateYear, dateMonth, dateDay);
+      break;
+      default:
+        // Open user file browser. User gives location of file to be created.
+        dialog.showSaveDialog({
+          defaultPath: dateString,
+          filters: [{
+            name: 'Database File',
+            extensions: ['db']
+          }, ]
+        }, function(fileLocation) {
+          console.log(fileLocation);
+          if (typeof(fileLocation) === 'undefined') {
+            console.log('Export Aborted');
+            return;
+          }
+          fs.readFile(appDatabaseFile, function(readErr, data) {
+            if (readErr) {
+              throw readErr;
+            }
+            fs.writeFile(fileLocation, data, function(writeErr) {
+              if (writeErr) {
+                throw writeErr;
+              }
+              showToast('Susbcriptions have been successfully exported');
+            });
+          })
+        });
+    }
+}
+/**
+ * Export the subscriptions database compatable with NewPipe.
+ *
+ * @return {Void}
+ */
+function exportNewpipeSubscriptions(dateYear, dateMonth, dateDay){
+
+  const dateString = 'newpipe-subscriptions-' + dateYear + '-' + dateMonth + '-' + dateDay;
+
   dialog.showSaveDialog({
     defaultPath: dateString,
     filters: [{
-      name: 'Database File',
-      extensions: ['db']
+      name: 'JSON',
+      extensions: ['json']
     }, ]
   }, function(fileLocation) {
     console.log(fileLocation);
     if (typeof(fileLocation) === 'undefined') {
       console.log('Export Aborted');
       return;
     }
-    fs.readFile(appDatabaseFile, function(readErr, data) {
-      if (readErr) {
-        throw readErr;
-      }
-      fs.writeFile(fileLocation, data, function(writeErr) {
-        if (writeErr) {
-          throw writeErr;
+    returnSubscriptions().then((result)=>{
+        let newpipe = {
+          app_version: "0.15.1",
+          app_version_int: 71,
+          subscriptions: []
         }
-        showToast('Susbcriptions have been successfully exported');
+        for (let i=0; i < result.length; i++) {
+          invidiousAPI('channels', result[i].channelId, {}, (iapi)=>{
```

Comment:
All of the information needed to make this file should already be saved in the user's database.  We shouldn't need to make an API call here.

Reply:
The reason i made api call is that Newpipe subs have a mix of both /channel/ & /user/ links, Api resolves exactly as opposed to the db. That's the reason. Let me know your opinion on this

Reply:
Will NewPipe accept the export file if we only use the `/channel/` links?  I don't know of a reason as to why it wouldn't.  If it accepts the file that way, let's go ahead and use the information already in the database.  If NewPipe requires the `/user/` links for certain channels then this can be left the way it is.

Reply:
Will check that out & report back to you soon

Reply:
Works!